app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/channels/DefaultChannelAdapter.kt

```diff
@@ -64,7 +66,18 @@ class DefaultChannelAdapter(
             ChannelTile.setChannelMarginByPosition(holder.itemView, context, position, itemCount)
             val tile = getItem(position)
             tile.setImage.invoke(imageView)
-            titleView.text = tile.title
+
+            // We handle Pocket tiles differently.
```

Comment:
It's reasonable with the way it is now and there's a time pressure for this, but we should consider polymorphism for `ChannelTile` instead of null checking:

```
when (tile) {
	is PocketTile -> // pocket stuff
	is PinnedTile -> // pinned_tile stuff
}
```

Reply:
I would take a different approach on this one, personally.  I would rename these from `titleView`, `pocketTitle`, and `pocketAuthor` to something generic like `aboveChannelText`, `belowChannelHeader`, `belowChannelText` (or some better version of all of that).  Then we can give `ChannelTile` `String?` fields to match, and it simply sets if present, or hides if null.

I feel really strongly that the presentation layer should not be conditionally styling based on the type of tile.  This approach would also give us a lot more flexibility to mix and match these options for future content.